File: F1ServerApp/src/app/data/participantviewdata.ts
import { ParticipantViewApiData } from "./participantdata_api";

export class ParticipantViewData
{
  // Fields
  participantDbId: number = 0;
  driverName: string = "";
  teamName: string = "";
  driverNationality: string = "";
  isHumanControlled: boolean = false;
  isMyTeam: boolean = false;
  carRaceNumber: number = 0;

  // Methods
  setParticipantApiData(participantData: ParticipantViewApiData)
  {
    if (participantData) {
      this.participantDbId = participantData.participantDbId;
      this.driverName = participantData.driverName;
      this.teamName = participantData.teamName;
      this.driverNationality = participantData.driverNationality;
      this.isHumanControlled = participantData.isHumanControlled;
      this.isMyTeam = participantData.isMyTeam;
      this.carRaceNumber = participantData.carRaceNumber;
    }
  }
}

File: F1Server.Data/ViewData/ParticipantViewData.cs
namespace F1Server.Data.ViewData;

/// <summary>
/// Class to view participants
/// </summary>
public class ParticipantViewData
{
    #region Properties

    /// <summary>
    /// Id of participant
    /// </summary>
    public long ParticipantDbId { get; set; }

    /// <summary>
    /// Driver name
    /// </summary>
    public string DriverName { get; set; }

    /// <summary>
    /// Nationality
    /// </summary>
    public string DriverNationality { get; set; }

    /// <summary>
    /// Name of team
    /// </summary>
    public string TeamName { get; set; }

    /// <summary>
    /// Is human controlled
    /// </summary>
    public bool IsHumanControlled { get; set; }

    /// <summary>
    /// Number of car
    /// </summary>
    public int CarRaceNumber { get; set; }

    /// <summary>
    /// Flag if this driver is in a custom created team
    /// </summary>
    public bool IsMyTeam { get; set; }

    #endregion // Properties
}